### PR TITLE
MB-5744 remediation g106 ssh host key violation

### PR DESCRIPTION
**Pull request by gidjin**
## Description

The static analysis review uncovered a G106 gosec warning in our code which was introduced by #5039. This is related to the SSH Host Key. This PR resolves it by reading the host key from environment variables set via chamber. 

## Reviewer Notes

There are tests for the session service, one of which as a real looking host key. However, this host key was generated specifically for the purpose of this test and is not used by any servers. Only the public key was saved. It was generated using the command `ssh-keygen -q -N "" -t ecdsa -f /tmp/ssh_host_ecdsa_key`

We will be testing the actual connection in the coming weeks so no need to test connecting to sycnada locally.

## Setup

Ensure things are still functioning

### Processing Reviewed Payment Requests, send to syncada FALSE

Setup

```sh
make db_dev_e2e_populate
make server_run
make bin/prime-api-client
```

Save payload.json
```json
{
  "body": {
    "sendToSyncada": false
  }
}
```

Trigger process payment request
```sh
prime-api-client --insecure support-reviewed-payment-requests --filename payload.json | jq .
```

```json
[
  {
    "eTag": "MjAyMC0xMi0wMlQyMDo0NTowMC4yNDg5Mlo=",
    "id": "cfd110d4-1f62-401c-a92c-39987a0b4228",
    "isFinal": false,
    "moveTaskOrderID": "7cbe57ba-fd3a-45a7-aa9a-1970f1908ae7",
    "paymentRequestNumber": "9188-5008-1",
    "status": "SENT_TO_GEX"
  }
]
```

### Verify fake host key

Double check that the fake host key in the test file does not match the real host key in chamber.

```sh
chamber env app-devlocal | grep SYNCADA_SFTP_HOST_KEY
```

### Verify real key is ready for exp and stg

Compare to key from devlocal above. Note key shouldn't be in prd, but only infra has access to prd anyway.

```sh
aws-vault exec transcom-gov-milmove-exp -- chamber env app-exp | grep SYNCADA_SFTP_HOST_KEY
aws-vault exec transcom-gov-milmove-stg -- chamber env app-stg | grep SYNCADA_SFTP_HOST_KEY
```

## Code Review Verification Steps

* [X] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [X] Request review from a member of a different team.
* [X] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-5744) for this change
* [this slack thread](https://ustcdp3.slack.com/archives/CP4UNF7H6/p1606774723183800?thread_ts=1606771930.183000&cid=CP4UNF7H6) explains more about the approach used.
* Original PR #5039 
* PR with relevant discussion #5373

### CODE REVIEW

File: pkg/cli/syncada_sftp.go [hunk 1]
```diff
@@ -1,34 +0,0 @@
-package cli
-
-import "github.com/spf13/pflag"
```
Comment (sojeri) [inline pkg/cli/syncada_sftp.go:3]: goodbye, pflag. you have your place, but for this time we decided to do it ourselves. at the end of the day, we're all just `os.getenv(<key>)` anyway.

File: pkg/services/invoice/syncada_sftp_sender_test.go [hunk 2]
```diff
@@ -101,8 +117,24 @@ func (suite *SyncadaSftpSenderSuite) TestSendToSyncadaSftp() {
...
 		os.Setenv("SYNCADA_SFTP_INBOUND_DIRECTORY", "/Dropoff")
+		// generated fake host key to pass parser used following command and only saved the pub key
+		//   ssh-keygen -q -N "" -t ecdsa -f /tmp/ssh_host_ecdsa_key
+		os.Setenv("SYNCADA_SFTP_HOST_KEY", "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBI+M4xIGU6D4On+Wxz9k/QT12TieNvaXA0lvosnW135MRQzwZp5VDThQ6Vx7yhp18shgjEIxFHFTLxpmUc6JdMc= fake@localhost")
 		sender, err := InitNewSyncadaSFTPSession()
 		suite.NoError(err)
 		suite.NotNil(sender)
 	})
+
+	suite.T().Run("constructor failes with invalid host key", func(t *testing.T) {
```
Comment (sojeri) [inline pkg/services/invoice/syncada_sftp_sender_test.go:128]:
```suggestion
	suite.T().Run("constructor fails with no host key", func(t *testing.T) {
		os.Setenv("SYNCADA_SFTP_PORT", "1234")
		os.Setenv("SYNCADA_SFTP_USER_ID", "FAKE_USER_ID")
		os.Setenv("SYNCADA_SFTP_IP_ADDRESS", "127.0.0.1")
		os.Setenv("SYNCADA_SFTP_PASSWORD", "FAKE PASSWORD")
		os.Setenv("SYNCADA_SFTP_INBOUND_DIRECTORY", "/Dropoff")
		os.Setenv("SYNCADA_SFTP_HOST_KEY", "")
		sender, err := InitNewSyncadaSFTPSession()
		suite.Error(err)
		suite.Nil(sender)
		suite.Equal("Failed to parse host key ssh: no key found", err.Error())
	})

	suite.T().Run("constructor fails with invalid host key", func(t *testing.T) {
```

**Review by sojeri — COMMENTED** [review]
the code looks good to me. I haven't tested it yet-- hence my lack of approval. will try to get to that before noon PST.

**Review by jneuner-mitre — COMMENTED** [review]
looks ok to me.  Only _suggestion_ is to change SYNCADA_SFTP_IP_ADDRESS to SYNCADA_SFTP_ADDRESS; looks like the framework will resolve DNS names as well.

**Review by reggieriser — COMMENTED** [review]
Did setup steps and verifications and all worked as described.

**Review by sojeri — APPROVED** [review]
:rocket:

**Pull request merged**